fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -841,10 +912,14 @@ private ScoredPlan handleNonSargables(@Nonnull ScoredPlan bestPlan,
             comparisonKey = getKeyForMerge(sort, comparisonKey);
             ScoredPlan intersectionPlan = planIntersection(intersectionCandidates, comparisonKey);
             if (intersectionPlan != null) {
-                if (intersectionPlan.unsatisfiedFilters.isEmpty()) {
-                    return intersectionPlan;
-                } else if (bestPlan.getNumNonSargables() > intersectionPlan.getNumNonSargables()) {
-                    bestPlan = intersectionPlan;
+                // number of sarged in-bindings in intersection plan can only be lower or equal to sarged in-bindings
+                // of best plan
+                if (bestPlan.getSargedInBindings().size() == intersectionPlan.getSargedInBindings().size()) {
```

Comment:
I'm struggling a bit with this assertion ("number of sarged in-bindings in intersection plan can only be lower or equal to sarged in-bindings of best plan"). Is it saying that the only kinds of intersection plans that we expect are those for which the same sets of `IN`-bindings are sargable on all legs, and that the `bestPlan` from before will either be on one of those indexes (and thus have the same number of sargable in bindings) or will be on some other index with more sargable `IN`s?

Even if it is true, do we still want to favor the `intersectionPlan` in this case? This seems somewhat analogous to the previous decision to favor sargable `IN`s over saragable predicates that used to be in this PR but was removed

Reply:
The intersection logic for finding the sargables intersects the sets of comparisons for all legs. Since the best plan is among those, the intersection set can only be at most that.

I reverted the code to be in line with the preferences for individual plan comparison i.e. in `isBetterThanOther()`